ios/RNGoogleSignin/RNGoogleSigninButtonManager.m

```diff
@@ -11,30 +12,35 @@ @implementation RNGoogleSigninButtonManager
 
 - (UIView *)view
 {
-  GIDSignInButton *button = [[GIDSignInButton alloc] init];
+  RNGoogleSignInButton *button = [[RNGoogleSignInButton alloc] init];
   button.colorScheme = kGIDSignInButtonColorSchemeLight;
   button.style = kGIDSignInButtonStyleStandard;
 
-//  [button removeTarget:nil action:NULL forControlEvents:UIControlEventTouchUpInside];
-//  [button addTarget:self action:@selector(buttonAction:) forControlEvents:UIControlEventTouchUpInside];
+  [button removeTarget:nil action:NULL forControlEvents:UIControlEventTouchUpInside];
+  [button addTarget:self action:@selector(onPress:) forControlEvents:UIControlEventTouchUpInside];
+
   return button;
 }
 
-RCT_CUSTOM_VIEW_PROPERTY(color, NSString, GIDSignInButton)
+RCT_CUSTOM_VIEW_PROPERTY(color, NSString, RNGoogleSignInButton)
 {
-  view.colorScheme = json ? [RCTConvert GIDSignInButtonColorScheme:json] :kGIDSignInButtonColorSchemeLight;
+  view.colorScheme = json ? [RCTConvert GIDSignInButtonColorScheme:json] : kGIDSignInButtonColorSchemeLight;
+}
 
+RCT_CUSTOM_VIEW_PROPERTY(size, NSString, RNGoogleSignInButton)
+{
+  view.style = json ? [RCTConvert GIDSignInButtonStyle:json] : kGIDSignInButtonStyleStandard;
 }
 
-RCT_CUSTOM_VIEW_PROPERTY(size, NSString, GIDSignInButton)
+RCT_EXPORT_VIEW_PROPERTY(onPress, RCTBubblingEventBlock)
```

Comment:
suggestion: we could not handle the onPress at all in the native code. Instead, the sign in button will be just a dummy component that will show up in the app if the user adds it. We will wrap it in a touchable, and set `pointerEvents="none"` so the native button will effectively ignore the onPress. I think this will allow handling everything from JS once, and not have extra ObjectiveC and java code.

Reply:
Um, yeah, maybe not use pointerEvents=none which is too radical but not specifing any on Press handler in native should work. I think one bridge round-trip is not a big problem and the transparency of having one onPress handler in JS is worthwhile.

Reply:
Now that I think about it, the roundrip has to be made anyway, regardless of whether we have the onPress code in native. Even like this, when user presses the button, RN has to go to JS to find out what to do and then go back to native to show the login dialog.

Reply:
just for the record, using Touchable* would not introduce any additional roundtrips, as I though we clarified on slack. @jozan 
 please feel free to explain where the additional roundtrips would come from if you believe they would be there. For now, I suggest to keep this as-is, to not bloat the PR with more stuff.

Reply:
My poor (and incorrect) wording has let me down. I meant to say that the current status is in this PR it would not introduce any additional roundtrips over the bridge regardless if the underlying button is Touchable* or native GoogleSignInButton. Let's keep this as-is. 👍